apps/linkedin-to-notion/src/tabs/onboarding.tsx

```diff
@@ -0,0 +1,169 @@
+import { ButtonPrimary, Card, Heading1, Heading2 } from 'design-system';
+
+import '~style.css'; // mandatory to inject tailwindcss styles
+
+import type { Session } from '@supabase/supabase-js';
+import { useEffect, useState } from 'react';
+import arrow from 'url:./../../assets/arrow.svg';
+import linkedin_logo from 'url:./../../assets/linkedin_logo.svg';
+import notion_logo from 'url:./../../assets/notion_logo.svg';
+import pin_icon from 'url:./../../assets/pin_icon.svg';
+
+import { sendToBackground } from '@plasmohq/messaging';
+import { useStorage } from '@plasmohq/storage/hook';
+
+import { supabase } from '~core/supabase';
+import { OnboardingStatus } from '~src/background/messages/users/services/user.service';
+import type { Tables } from '~src/background/types/supabase';
+import { BasicLayout } from '~src/components/Layouts/BasicLayout';
+import { Confetti } from '~src/components/Layouts/Confetti';
+import { handleOAuthLogin } from '~src/contents/handleOAuthLogin';
+import { routes } from '~src/routes';
+
+export const OnboardingPage = () => {
+  const [numberOfTasks, setNumberOfTasks] = useState<number>(3);
+  const [user, setUser] = useStorage<Tables<'users'> | null>('user');
+  const [session, setSession] = useStorage<null | Session>('session');
+
+  useEffect(() => {
+    const initSession = async () => {
+      const { data } = await supabase.auth.getSession();
+      setSession(data.session);
+    };
+
+    const initUser = async () => {
+      const userData = await sendToBackground<
+        {
+          authenticatedUserId: string;
+        },
+        Tables<'users'>
+      >({
+        name: 'users/resolvers/getOrCreateUserWithAuthenticatedUserId',
+        body: { authenticatedUserId: session.user.id },
+      });
+      setUser(userData);
+    };
+
+    if (!session || session?.expires_at < Date.now()) {
```

Comment:
```suggestion
    if (!session) {
```
It's not necessary to check the expiration date because [it's only available in pro plans](https://supabase.com/docs/guides/auth/sessions#limiting-session-lifetime-and-number-of-allowed-sessions-per-user).

Furthermore, the current condition **always resolves to true** because the "expires_at" timestamp is set in such a way that it's not a UTC timestamp (take `new Date(sessions.expires_at).toISOString()` to make sure of it) and it's always inferior to Date.now().

Because of that, we trigger an endless loop and raise a MAX_WRITE_ERROR that's already been encountered in #67 .